bin/util.py

```diff
@@ -466,6 +474,24 @@ def read_yaml_settings(path):
     return settings
 
 
+# Writing a yaml file only works when ruamel.yaml is loaded. Check if `has_ryaml` is True before using.
+def write_yaml(data, path):
+    ryaml.dump(
+        data,
+        path,
+        # Remove spaces at the start of each (non-commented) line, caused by the indent configuration.
+        # This is only needed when the YAML data is a list of items, like in the problems.yaml file.
+        # See also: https://stackoverflow.com/a/58773229
+        transform=(
```

Comment:
I suspect this transform is broken when the data is not actually a list. Can you test, or just do `if instanceof(data, list): dump with transform else: dump without transform`

Reply:
I tested this with `bapctools generate --add-manual`, which rewrites the `generators/generators.yaml`. This yaml file is a `dict`, not a `list`, so `transform=None` in that case. :slightly_smiling_face: 

The code that you propose does exactly the same as the inline conditional that it currently is. Although I understand that the formatting of the code does not make that very clear, so I added some parentheses, hopefully that helps :smile:

Reply:
Oh I completely missed the if statement inside there. Should be good now